File: questions/0056. Merge Intervals/merge.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        res = []
        for ele in sorted(intervals, key = lambda x:x[0]):
            if res and ele[0] <= res[-1][1]:
                res[-1][1] = max(ele[1], res[-1][1])
            else:
                res.append(ele)
        return res
        

File: questions/0056. Merge Intervals/merge.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end(), [](vector<int>& a, vector<int>& b){return a[0]< b[0];});
        vector<vector<int>> res;
        res.push_back(intervals[0]);
        for(vector<int>& ele: intervals){
            if(ele[0] <= res.back()[1]){
                res.back()[1] = max(res.back()[1], ele[1]);
            }else{
                res.push_back(ele);
            }
        }
        return res;
    }
};
